### DISCUSSION TITLE
Using 'if' in GitHub actions workflow

### DISCUSSION THREAD

**Discussion by VahilaK** [community/community#27105] [Actions]
<p>I am trying to run a job on different runners using matrix configuration. Runners names are given as matrix parameters. I want to run a step only for a particular runner</p>
<p>I am using a if statement checking the runners name but the step is run even when the if is false. Example code is:</p>
<pre><code class="lang-auto">name: Testing if
on: push
jobs:
    Build_testing_with_if:
      strategy:
            matrix:
                platform: [ubuntu-latest, ubuntu-18.04]
      runs-on: ${{ matrix.platform }}
      steps:
        - name: Check out repository
          uses: actions/checkout@v2
          
        - if: ${{matrix.platform}} == 'ubuntu-latest'
          name: Print          
          run: echo "matrix platform is" ${{matrix.platform}}
</code></pre>
<p>I expect the print to happen when running on ‘ubuntu-latest’ only but it is printed for both runners. How can I run the step only for one runner?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The problem is that you have only <code>matrix.platform</code> wrapped in <code>${{ ... }}</code>. That marks it as an expression to evaluate, and the rest of the line is implicitly treated as text. And a non-empty string evaluates to “true”.</p>
<p>So, either wrap the whole expression you want to be evaluated:</p>
<pre><code class="lang-yaml">- if: ${{ matrix.platform == 'ubuntu-latest' }}
</code></pre>
<p>Or (special rule for <code>if</code>) just leave the <code>${{ ... }}</code> and the whole thing will be treated as an expression:</p>
<pre><code class="lang-yaml">- if: matrix.platform == 'ubuntu-latest'
</code></pre>
